datafusion/functions/src/core/planner.rs

```diff
@@ -38,3 +40,28 @@ impl UserDefinedSQLPlanner for CoreFunctionPlanner {
         Ok(PlannerResult::Planned(named_struct().call(args)))
     }
 }
+
+#[derive(Default)]
+pub struct CreateStructPlanner;
+
+impl UserDefinedSQLPlanner for CreateStructPlanner {
+    fn plan_create_struct(&self, args: Vec<Expr>) -> Result<PlannerResult<Vec<Expr>>> {
+        Ok(PlannerResult::Planned(Expr::ScalarFunction(
+            ScalarFunction::new_udf(crate::core::r#struct(), args),
```

Comment:
I think we can call `named_struct` with default key `c1, c2, ... cn`, then we can probably remove whole the struct function.

Reply:
Great idea @jayzhan211 . I will see if i can incorpoate this change in the PR or have a new for this.